src/Customer.php

```diff
@@ -354,4 +354,72 @@ public function createContact(array $data = [])
 
         return new Contact($result, $client);
     }
+
+    /**
+     * Find all customer notes in a customer
+     *
+     * @param  array $options
+     * @return CollectionWithCursor
+     */
+    public function customer_notes(array $options = [])
+    {
+        $client = $this->getClient();
+        $result = $client->send("/v1/customers/".$this->uuid."/notes", "GET", $options);
+
+        return CustomerNote::fromArray($result, $client);
+    }
+
+    /**
+     * Creates a customer note from the customer.
+     *
+     * @param  array $data
+     * @return CustomerNote
+     */
+    public function createCustomerNote(array $data = [])
+    {
+        $client = $this->getClient();
+        $result = $client->send("/v1/customers/".$this->uuid."/notes", "POST", $data);
+
+        return new CustomerNote($result, $client);
+    }
+
+    /**
+     * Retrieves a customer note from the customer.
+     * @param  string $note_uuid
+     * @return CustomerNote
+     */
+    public function retrieveCustomerNote(string $note_uuid)
+    {
+        $client = $this->getClient();
+        $result = $client->send("/v1/customers/".$this->uuid."/notes/".$note_uuid, "GET");
+
+        return new CustomerNote($result, $client);
+    }
+
+    /**
+     * Updates a customer note from the customer.
+     * @param  string $note_uuid
+     * @param  array $data
+     * @return CustomerNote
+     */
+    public function updateCustomerNote(string $note_uuid, array $data = [])
+    {
+        $client = $this->getClient();
+        $result = $client->send("/v1/customers/".$this->uuid."/notes/".$note_uuid, "PATCH", $data);
+
+        return new CustomerNote($result, $client);
+    }
+
+    /**
+     * Deletes a customer note from the customer.
+     * @param  string $note_uuid
+     * @return CustomerNote
+     */
+    public function deleteCustomerNote(string $note_uuid)
+    {
+        $client = $this->getClient();
+        $result = $client->send("/v1/customers/".$this->uuid."/notes/".$note_uuid, "DELETE");
+
+        return new CustomerNote($result, $client);
```

Comment:
The result for this is `{}` from the API and it seems that you're returning a note here. It's a little inconsistent with how the HTTP API works